lib/test/t_TensorProductExperiment_std.cxx

```diff
@@ -0,0 +1,234 @@
+//                         -*- C++ -*-
+/**
+ *  @brief The test file of class TensorProductExperiment for standard methods
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/OT.hxx"
+#include "openturns/OTtestcode.hxx"
+
+using namespace OT;
+using namespace OT::Test;
+
+// Check if two int are equal
+void assertEqual(int value1, int value2)
+{
+  if (value1 != value2)
+  {
+  throw TestFailed(OSS() << "Value " << value1 << " is not equal to " << value2);
+  }
+}
+
+// Compare two points, according to lexicographic order
+/** Returns -1 if node_1 < node_2, 
+ *           0 if node1 == node_2,
+ *           1 otherwise.
+ */
+UnsignedInteger comparePoints(const Point point1, const Point point2)
+{
+  const UnsignedInteger dimension(point1.getDimension());
+  if (!(point2.getDimension() == dimension))
+    throw InvalidArgumentException(HERE) << "Error: Dimension of point 1 is " << dimension << " but dimension of point 2 is " << point2.getDimension();
+  UnsignedInteger comparison = 0;
+  for (UnsignedInteger k = 0; k < dimension; ++k)
+  {
+    if (point1[k] < point2[k])
+    {
+      comparison = -1;
+      break;
+    }
+    if (point1[k] > point2[k])
+    {
+      comparison = 1;
+      break;
+    }
+  }
+  return comparison;
+}
+
+// Simultaneously sort the nodes and weights
+void sortNodesAndWeights(Sample & nodes, Point & weights)
+{
+  OStream fullprint(std::cout);
+  fullprint << "+ Simultaneously sort the nodes and weights" << std::endl;
+  const UnsignedInteger size(weights.getDimension());
+  const UnsignedInteger dimension(nodes.getDimension());
+  fullprint << "size = " << size << std::endl;
+  fullprint << "dimension = " << dimension << std::endl;
+  UnsignedInteger isIGreaterThanJ;
```

Comment:
Ok. I rename this. Notice, however, that this particular code may go out of the source if I am successful at creating a `Sample.argsort` method. This is the topic I am working on in the last few hours: 
https://github.com/mbaudin47/openturns/blob/TestSortSample/lib/test/t_Sample_sort.cxx
This method would be very handy to have in order to create and test the 2 or 3 Experiment classes that I have in mind. It could fit, at least, in the OTtest code, so that we could test DOE without being bothered on the order of the points within the sample.

Reply:
I updated the code and renamed the variable.